### PR TITLE
Unfork spf13/{cobra, pflag}

### LINKED ISSUE #510: `nerdctl run` argument parsing order potentially broken?
From `nerdctl run --help`:

```
Usage:
  nerdctl run IMAGE [COMMAND] [ARG...] [flags]
```

From `docker run --help`:

```
Usage:  docker run [OPTIONS] IMAGE [COMMAND] [ARG...]
```

We are trying to run a container in a script with a complex set of options passed to the command but because of this ordering it's not clear that `nerdctl` can safely disambiguate between arguments it should process and arguments for the spawned container.

The docker equivalent command looks like this:

```
docker run --rm --mount type=bind,source=A_PATH,target=A_PATH --env PGHOST=host.lima.internal --env PGPORT=5432 --env PGUSER=postgres --env PGPASSWORD=password -it postgres:12.3 pg_dump --schema-only --no-privileges --no-owner --file A_FILE db_name
```

If I attempt to use the argument order in the help, it of course tries to pass `run` arguments like `--rm` and `--env` to the `pg_dump` binary in the container and fails. If someone can confirm this is unexpected, I may be able to take a look at resolving it.

### LINKED ISSUE #522: Unknown flags should not be ignored
```console
$ nerdctl version --some-dummy-unimplmented-flag
Client:
 Version:       v0.13.0-23-g24168f0
 Git commit:    24168f059511d9296ede4347d83dc8f3f1b2b506

Server:
 containerd:
  Version:      v1.6.0-beta.2
  GitCommit:    bd81f8a39d85bd7275757c719e08e69a5e997132
```

nerdctl should print warning (or just error out) when unknown flags were specified.

### PR DESCRIPTION
**Pull request by AkihiroSuda**
(See the commit message)

Fix #522
Fix #510

---

I'll release v0.14.0 after merging this

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
cc @robberphex

**Comment by AkihiroSuda** [comment]
```
$ nerdctl images
panic: unable to redefine 'a' shorthand in "images" flagset: it's already used for "all" flag
```

😕

**Comment by AkihiroSuda** [comment]
```
$ nerdctl compose logs --help
panic: unable to redefine 'f' shorthand in "logs" flagset: it's already used for "follow" flag
```

**Comment by robberphex** [comment]
It looks work to me, and thanks for fixing cli related issues.

File: cmd/nerdctl/main.go [hunk 1]
```diff
@@ -312,3 +277,58 @@ func unknownSubcommandAction(cmd *cobra.Command, args []string) error {
...
+		value = envV
+	}
+	aliasesUsage := fmt.Sprintf("Alias of --%s", name)
+	p := new(string)
+	flags := cmd.Flags()
+	flags.StringVar(p, name, value, usage)
+	for _, a := range aliases {
+		if len(a) == 1 {
+			// pflag doesn't support short-only flags, so we have to register long one as well here
+			flags.StringVarP(p, "dummylongflag-"+a, a, value, aliasesUsage)
```
Comment (ktock) [inline cmd/nerdctl/main.go:296]: `StringVarP` seems to support shorthand flags https://pkg.go.dev/github.com/spf13/pflag#FlagSet.StringVarP .
And `flags.StringVarP` called in `AddPersistentStringFlag` doesn't seem to have `dummylongflag`.

Comment (AkihiroSuda) [reply cmd/nerdctl/main.go]: Fixed

**Review by ktock — APPROVED** [review]

**Pull request merged**